packages/mdl-drawer/_mixins.scss

```diff
@@ -0,0 +1,152 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@import "./functions";
+@import "mdl-theme/mixins";
+@import "mdl-typography/mixins";
+
+/* All variables are local, so disable pattern checking. */
+/* stylelint-disable scss/dollar-variable-pattern */
+
+@mixin mdl-drawer-dir_($property, $value) {
+  $directions: (
+    ltr: (
+      "start": "left",
+      "end": "right"
+    ),
+    rtl: (
+      "start": "right",
+      "end": "left"
+    )
+  );
+
+  #{mdl-drawer-replace($property, map-get($directions, ltr))}: $value;
+
+  [dir="rtl"] & {
+    #{mdl-drawer-replace($property, map-get($directions, ltr))}: auto;
+    #{mdl-drawer-replace($property, map-get($directions, rtl))}: $value;
+  }
+}
+
+@mixin mdl-drawer-toolbar-spacer_() {
+  &__toolbar-spacer {
+    display: flex;
+    position: relative;
+    flex-direction: row;
+    flex-shrink: 0;
+    align-items: flex-center;
+    height: 56px;
+    padding: 16px;
+    border-bottom: 1px solid rgba(0, 0, 0, .12);
+    box-sizing: border-box;
+
+    @include mdl-theme-dark(quote(#{&})) {
+      border-bottom: 1px solid rgba(255, 255, 255, .12);
+    }
+
+    /* TODO(sgomes): replace with global breakpoints when we have them */
+    @media (min-width: 600px) {
+      height: 64px;
+    }
+  }
+}
+
+@mixin mdl-drawer-header_() {
+  /* Use aspect ratio trick to maintain 16:9 aspect ratio on the header */
+  &__header {
+    position: relative;
+  }
+
+  &__header::before {
+    display: block;
+    padding-top: 9 / 16 * 100%;
+    content: "";
+  }
+
+  &__header-content {
+    display: flex;
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    align-items: flex-end;
+    padding: 16px;
+    box-sizing: border-box;
+  }
+}
+
+@mixin mdl-drawer-list_() {
+  .mdl-list-group,
+  .mdl-list {
+    padding-right: 0;
+    padding-left: 0;
+  }
+
+  .mdl-list-item {
+    position: relative;
+    padding: 0 16px;
+    outline: none;
+    color: inherit;
+    text-decoration: none;
+
+    @include mdl-typography(body2);
+  }
+
+  .mdl-list-item__start-detail {
+    @include mdl-theme-prop(color, text-icon-on-background);
+
+    color: rgba(0, 0, 0, .54);
+
+    @include mdl-theme-dark(#{&}) {
+      color: rgba(255, 255, 255, .54);
+    }
+  }
+
+  /* stylelint-disable selector-no-qualifying-type */
+  &--selected.mdl-list-item,
+  &--selected.mdl-list-item .mdl-list-item__start-detail {
+    @include mdl-theme-prop(color, primary);
+  }
+  /* stylelint-enable selector-no-qualifying-type */
+
+  .mdl-list-item::before {
```

Comment:
nit: could we add a `TODO:` here to revisit and make interactive once we have interactive lists (e.g. #4614 )